Update polynomial_spectral_phase.py

mjvstreeter opened the pull request:
Added optional delay keyword argument to PolynomialSpectralPhase optical element. Adds capability to delay pulse in time.

mjvstreeter committed:
Update polynomial_spectral_phase.py

Added optional delay input keyword argument.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mjvstreeter committed:
Update polynomial_spectral_phase.py

Fixed error in description. Pulse moves forward in time for positive delay.

rob-shalloo commented on lasy/optical_elements/polynomial_spectral_phase.py:
```diff
@@ -33,15 +33,18 @@ class PolynomialSpectralPhase(OpticalElement):
     fod : float (in s^4), optional
         Fourth-order Dispersion (by default: ``fod=0``).
+    delay : float (in s), optional
+        Delay (by default: ``delay=0``). Positive delay moves pulse forward in time.
```
```suggestion
        Group Delay (by default: ``delay=0``). Positive delay moves pulse forward in time.
```

mjvstreeter replied:
Thanks for the suggestions. I have incorporated them and also clarified what group delay means.

rob-shalloo commented:
Hey Matt, thanks for the PR! I think this is a great addition and was certainly missing before.

Given that gdd,tod,fod and delay all default to 0, I would suggest reordering the arguments to a more natural order eg: delay, gdd, tod and fod.

Additionally, I would suggest updating the latex equation in the docstring to include the delay term.

I've also added a suggested change to update the name from `Delay` to `Group Delay` in the description of the argument.

mjvstreeter committed:
Update polynomial_spectral_phase.py

Updated order of arguments to make them go in increasing order of omega. Also updated description of delay and latex equation for spectral phase.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rob-shalloo committed:
Update polynomial_spectral_phase.py

removed double description of `delay` parameter and moved `omega0` to first entry

rob-shalloo commented:
I've just removed double description of the `delay` parameter and moved `omega0` to first entry in the parameter list as it's the only required argument.

rob-shalloo commented:
Discussed over zoom, Matt to add test of delay to CI before review

mjvstreeter committed:
Update test_polynomial_spectral_phase.py

Added test for new delay argument to polynomial spectral phase optic.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mjvstreeter commented:
Added the test of delay. 
Also, noticed that grid parameters defined before functions are never used (in case you want to do anything about that).

MaxThevenet reviewed (approved):
Thanks for this PR!
